prov/tcp/src/tcpx_progress.c

```diff
@@ -204,30 +169,30 @@ static int process_srx_entry(struct tcpx_xfer_entry *rx_entry)
 		return ret;
 
 	if (ret) {
-		FI_WARN(&tcpx_prov, FI_LOG_DOMAIN,
+		FI_WARN(&tcpx_prov, FI_LOG_EP_DATA,
 			"msg recv Failed ret = %d\n", ret);
 
-		tcpx_ep_shutdown_report(rx_entry->ep,
-					&rx_entry->ep->util_ep.ep_fid.fid);
+		/* Is this check needed? srx did not have it */
+		if (ret == -FI_ENOTCONN) {
+			tcpx_ep_shutdown_report(rx_entry->ep,
+				&rx_entry->ep->util_ep.ep_fid.fid);
+		}
 	}
 
-	if ((ntohs(rx_entry->hdr.base_hdr.flags) &
-	     OFI_DELIVERY_COMPLETE) && !ret) {
+	/* srx included && !ret as part of check */
+	if (ntohs(rx_entry->hdr.base_hdr.flags) & OFI_DELIVERY_COMPLETE) {
```

Comment:
If the ret is non-zero, its better to shutdown the connection and not send the response. So that !ret check is needed